test/new-e2e/utils/e2e/e2e.go

```diff
@@ -91,38 +97,69 @@ func NewSuite[Env any](stackName string, stackDef *StackDefinition[Env], options
 	return &testSuite
 }
 
+func DevMode[Env any]() func(*Suite[Env]) {
+	return func(suite *Suite[Env]) {
+		suite.devMode = true
+	}
+}
+
+func SkipDeleteOnFailure[Env any]() func(*Suite[Env]) {
+	return func(suite *Suite[Env]) {
+		suite.skipDeleteOnFailure = true
+	}
+}
+
+// Env returns the current environment.
+// In order to improve the efficiency, this function behaves as follow:
+//   - It creates the default environment if no environment exists. It happens only during the first call of the test suite.
+//   - It restores the default environment if UpdateEnv was not already be called during this test.
+//     This avoid having to restore the default environment for each test even if UpdateEnv immedialy
+//     overrides this environment.
+func (suite *Suite[Env]) Env() *Env {
+	if suite.env == nil || !suite.isUpdateEnvCalledInThisTest {
+		suite.UpdateEnv(suite.defaultStackDef)
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Why not calling it in `AfterTest` if `!suite.isUpdateEnvCalledInThisTest == true`, so that you clean up after running a test ? 

This way you would not have to keep a single test state within the suite. Should we `panic` to stop the suite and skip `TearDown` on test failure ?

Reply:
Something like 
![image](https://user-images.githubusercontent.com/45568537/234268047-c6f29589-5e50-400c-9d45-b1bc56e0c687.png)

where each test is independent, as much as possible

Reply:
Even calling `panic` in `AfterTest` doesn't prevent the other tests to be run. The only way to stop tests is to call `os.Exit()` but then you miss all the test error messages. I will add a comment to explain that.